projects/pgai/db/sql/idempotent/014-semantic-catalog.sql

```diff
@@ -0,0 +1,1004 @@
+    
+    -- grab the catalog id
+    select c.id into strict _catalog_id
+    from ai.semantic_catalog c
+    where c.catalog_name = _catalog_name
+    ;
+    
+    if _embedding_name is null then
+        select 'emb' ||
+        greatest
+        ( count(*)::int4
+        , max((regexp_match(e.embedding_name, '[0-9]+$'))[1]::int4)
+        ) + 1
+        into strict _embedding_name
+        from ai.semantic_catalog_embedding e
+        where e.semantic_catalog_id = _catalog_id
+        ;
+    end if;
+    
+    insert into ai.semantic_catalog_embedding (semantic_catalog_id, embedding_name, config)
+    values (_catalog_id, _embedding_name, _config)
+    returning * into strict _embedding
+    ;
+    
+    -- add the columns
+    foreach _tbl in array array['obj', 'sql', 'fact']
+    loop
+        _sql = format
+        (
+        $sql$
+            alter table ai.semantic_catalog_%s_%s add column %s @extschema:vector@.vector(%s)
+        $sql$
+        , _tbl
+        , _catalog_id
+        , _embedding_name
+        , _dims
+        );
+        raise debug '%', _sql;
+        execute _sql;
+    end loop;
+    
+    perform ai._semantic_catalog_make_triggers(_catalog_id);
+    
+    return _embedding;
+end;
+$func$ language plpgsql volatile security invoker
+set search_path to pg_catalog, pg_temp
+;
+
+-------------------------------------------------------------------------------
+-- sc_drop_embedding
+create or replace function ai.sc_drop_embedding
+( embedding_name name
+, catalog_name text default 'default'
+) returns void
+as $func$
+declare
+    _embedding_name name = sc_drop_embedding.embedding_name;
+    _catalog_name text = sc_drop_embedding.catalog_name;
+    _embedding ai.semantic_catalog_embedding;
+    _catalog_id int4;
+    _tbl text;
+    _sql text;
+begin
+
+    select c.id into strict _catalog_id
+    from ai.semantic_catalog c
+    where c.catalog_name = _catalog_name
+    ;
+
+    delete from ai.semantic_catalog_embedding e
+    where e.semantic_catalog_id = _catalog_id
+    and e.embedding_name = _embedding_name
+    returning * into strict _embedding
+    ;
+    
+    -- drop the columns
+    foreach _tbl in array array['obj', 'sql', 'fact']
+    loop
+        _sql = format
+        (
+        $sql$
+            alter table ai.semantic_catalog_%s_%s drop column %s
+        $sql$
+        , _tbl
+        , _catalog_id
+        , _embedding_name
+        );
+        raise debug '%', _sql;
+        execute _sql;
+    end loop;
+    
+    perform ai._semantic_catalog_make_triggers(_catalog_id);
+end;
+$func$ language plpgsql volatile security invoker
+set search_path to pg_catalog, pg_temp
+;
+
+-------------------------------------------------------------------------------
+-- create_semantic_catalog
+create or replace function ai.create_semantic_catalog
+( catalog_name text default 'default'
+, embedding_name name default null
+, embedding_config jsonb default ai.embedding_sentence_transformers()
```

Comment:
is this the default used by Python? Shouldn't we just have the python determine the default?